src/components/ReviewVote.tsx

```diff
@@ -0,0 +1,60 @@
+import "../styles/VoteButtons.css";
+import { useEffect, useState } from "react";
+import { fetchSingleReview, patchVotes } from "../utils/api";
+import { ReviewVotesProps } from "../interfaces/review.interface";
+
+const ReviewVote = ({ votes, review_id }: ReviewVotesProps): JSX.Element => {
+  const [vote, setVotes] = useState<number>(votes || 0);
+  const [hasVotedUp, setHasVotedUp] = useState<boolean>(false);
+  const [hasVotedDown, setHasVotedDown] = useState<boolean>(false);
+
+  const handleVoteUp = async () => {
+    setVotes((prevVotes) => prevVotes + 1);
+    setHasVotedUp(true);
+    await patchVotes(review_id, 1);
+  };
+
+  const handleVoteDown = async () => {
+    setVotes((prevVotes) => prevVotes - 1);
+    setHasVotedDown(true);
+    await patchVotes(review_id, -1);
+  };
```

Comment:
Nice optimistic rendering - however, what happens if I dont have an internet connection/the request fails and click the button?

At the moment, it looks like all has gone well, until you refresh the page. 

Could you handle this occasion and make sure theres some feedback to let us know something has gone wrong?

Reply:
@popatre Cheers! I've updated now - thanks for the feedback :)